packages/shared/lib/core/router/subrouters/onboarding/network-setup-router.ts

```diff
@@ -20,13 +19,13 @@ export class NetworkSetupRouter extends Subrouter<NetworkSetupRoute> {
         const currentRoute = get(this.routeStore)
         switch (currentRoute) {
             case NetworkSetupRoute.ChooseNetwork: {
-                const networkType = _onboardingProfile?.networkType ?? NetworkType.Devnet
-                if (networkType === NetworkType.PrivateNet) {
-                    nextRoute = NetworkSetupRoute.SetupPrivateNetworkConnection
-                    break
-                } else {
+                const network = _onboardingProfile?.network
+                if (network) {
                     this.parentRouter.next()
                     return
+                } else {
+                    nextRoute = NetworkSetupRoute.SetupPrivateNetworkConnection
+                    break
```

Comment:
Might be better to be explicit here about the `type` being `Custom`

Reply:
Potentially but if we already have a network for some reason then why would we want to go to the form to add it in the router? But yeah that could cause issues with back button etc. So I can change it.